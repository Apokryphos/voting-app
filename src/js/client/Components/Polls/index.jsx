import React from 'react';
import PropTypes from 'prop-types';
import Poll from '../Poll';

function Polls(props) {
  const polls = props.polls.map(p => (
    <Poll key={p._id} id={p._id} question={p.question} choices={p.choices} />
  ));

  return <div>{polls}</div>;
}

Polls.propTypes = {
  polls: PropTypes.arrayOf(PropTypes.object),
};

Polls.defaultProps = {
  polls: [],
};

export default Polls;
