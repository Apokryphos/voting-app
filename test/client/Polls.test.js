import PollData from '../data/poll-data';
import Polls from 'Components/Polls';
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

describe('Polls rendering', () => {
  const pollData = PollData.valid();
  pollData._id = '59fbcdb370318a44f5f911ea';

  const pollsData = [pollData];

  it('Renders OK', () => {
    const component = ReactTestRenderer.create(<Polls polls={pollsData} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
