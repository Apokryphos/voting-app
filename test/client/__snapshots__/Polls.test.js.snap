// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Polls rendering Renders OK 1`] = `
<div>
  <div>
    <h1>
      What's your favorite hot dog?
    </h1>
    <ul>
      <li>
        Attman's Deli
      </li>
      <li>
        Ball Park Franks
      </li>
      <li>
        Dachshund
      </li>
      <li>
        Esskay
      </li>
      <li>
        Hebrew National
      </li>
      <li>
        Nathan's Famous
      </li>
    </ul>
  </div>
</div>
`;
